crates/optimism/consensus/src/beacon.rs

```diff
@@ -0,0 +1,188 @@
+//! Optimism beacon consensus rule set.
+
+#![cfg(feature = "std")]
```

Comment:
we can likely make this fully no-std if we do some prep first on these

https://github.com/paradigmxyz/reth/blob/28fde3fbf5c6803ded52aabca4b57845dec45ad3/crates/optimism/consensus/Cargo.toml#L22-L25

Reply:
hm, the issue is `std::time::SystemTime`, wdym?

Reply:
ah I found this now https://github.com/paradigmxyz/reth/issues/13683#issuecomment-2573956788